### ISSUE TITLE
Support using existing Auth0Client to improve performance

### ISSUE THREAD

**Issue by sijakret** [auth0/auth0-react#687]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/auth0-react#readme), [Examples](https://github.com/auth0/auth0-react/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-react/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-react/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-react/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

Currently, in our react application, authentication is only kicked of after the first react render call (i.e. when the [context is instantiated](https://github.com/auth0/auth0-react#configure-the-sdk)).

To improve bootstrapping of our react app we want to start authentication BEFORE react even starts rendering (or is even loaded). This idea has been uttered before (https://github.com/auth0/auth0-react/pull/278#issuecomment-923053283) and there exists a document with reasoning against it here: https://gist.github.com/adamjmcgrath/0ed6a04047aad16506ca24d85f1b2a5c
In our case we are also NOT interested in having multiple clients but are interested in the ability to **immediately start checking the session** right after the auth0client code is available and **not only when the first react render call is happening**?

Is this possible?
* If so: how? :)
* If not: Are there plans to enable it?

Thanky you for any input!

### Describe the ideal solution

Possible solutions:
1) provide an imperative API to either provide an existing Auth0Client to the context provider
2) provide a way to configure + trigger authentication before react context+context provider are even created.

**Comment by frederikprijck** [comment]
Thanks for reaching out. As you mention yourself, this has been asked for before and we still have no plans to support this.

I do want to call out the fact that our Auth0-React SDK is a tiny wrapper around Auth0-SPA-JS, that is build around the provider and context API for react.

If this is not what you want (which is the case, as you want to exit the Auth0Provider way of instatiating the client), it's perfectly find to use Auth0-SPA-JS directly and the lack of passing your own Auth0Client shouldn't block you. You can have a look at our Auth0-React SDK and rebuild whatever you want around Auth0-SPA-JS yourself to better suit your needs.

So long story short:

- It is not possible using Auth0-React, as this is a thin and opiniated Auth0-SPA-JS implementation for React.
- There are no plans to enable support for this, because it is already possible when using Auth0-SPA-JS directly.

Sure, using Auth0-SPA-JS directly can feel a bit complicated, but we encourage that for everyone that has more complicated use-cases as what our Auth0-React SDK offers.

**Comment by sijakret** [comment]
got it, thanks for the explanation!
